synapse/storage/data_stores/main/cache.py

```diff
@@ -66,7 +71,24 @@ def get_all_updated_caches_txn(txn):
         )
 
     def process_replication_rows(self, stream_name, instance_name, token, rows):
-        if stream_name == "caches":
+        if stream_name == "events":
+            self._stream_id_gen.advance(token)
+            for row in rows:
+                self._process_event_stream_row(token, row)
```

Comment:
moving this in here runs the risk of changing the order in which we carry out the operations we do when processing a replication row (since it basically relies on the MRO of `process_replication_rows`. Have you had a think about that?

Reply:
All the store stuff should just be invalidating caches or advancing tokens so the order doesn't matter (certainly I don't think we've thought about MRO before). The store `process_replication_rows` gets called before we otherwise handle rows too, so I think that is fine